analysis/src/main/scala/org/polystat/odin/analysis/unjustifiedassumptions/Analyzer.scala

```diff
@@ -47,20 +49,28 @@ object Analyzer {
       // println(after.body.toEOPretty)
       // println("==================================================")
       for {
-        methodsBefore <- getMethodsInfo("before", infoBefore.allMethods)
-        methodsAfter <- getMethodsInfo("after", infoAfter.allMethods)
+        methodsBefore <-
+          EitherT
+            .fromEither[F](getMethodsInfo("before", infoBefore.allMethods))
+        methodsAfter <-
+          EitherT.fromEither[F](getMethodsInfo("after", infoAfter.allMethods))
 
         res1 <-
-          processMethod("before", before, methodName, methodsBefore.keySet)
-        res2 <- processMethod("after", after, methodName, methodsAfter.keySet)
+          EitherT.fromEither[F](
+            processMethod("before", before, methodName, methodsBefore.keySet)
+          )
+        res2 <- EitherT.fromEither[F](
+          processMethod("after", after, methodName, methodsAfter.keySet)
+        )
         res <-
-          checkImplication(
+          checkImplication[F](
             methodName,
             res1,
             methodsBefore,
             res2,
             methodsAfter,
-            name => s"Method $name is not referentially transparent"
+            name =>
+              s"Method $name is not safe to inline: doing so may break the behaviour of subclasses!"
```

Comment:
@Leosimetti is there an easy way to change the error message to include the name of the object where this method was detected? Using `info.name` gives the name is the subclass. For example, for test:
```scala
       """|[] > c
           |  [self v] > l
           |    assert (v.less 5) > @
           |  [self v] > m
           |    self.l self v > @
           |  [self v] > n
           |    v > @
           |
           |[] > m
           |  c > @
           |  [self v] > l
           |    v > @
           |  [self v] > n
           |    self.m self v > @
           |""".stripMargin
```
the output is 
```
"Method m in object m is not safe to inline: doing so may break the behaviour of subclasses!"
```
but should be:
```
"Method m in object c is not safe to inline: doing so may break the behaviour of subclasses!"
```